### DISCUSSION TITLE
List of installed features or conditional bean

### DISCUSSION THREAD

**Discussion by b0ro** [quarkusio/quarkus#41382] [Q&A] [answered]
Hello,
I'm trying to write an extension with some custom health checks. I'm using: 

- https://quarkus.io/guides/smallrye-health and 
- https://github.com/eclipse/microprofile-health/tree/main. 

I want to enable my health checks if given extension or bean is present. In Spring I would use @ConditionalOnBean and @ConditionalOnProperty. 

I know that quarkus has @IfBuildProperty but is there anything to check if given bean is present or if extension is loaded? It's a build time feature and quarkus always logs all installed features so the info is there. I just don't know how to access it programmatically in a extension.

**Comment by mschorsch** [comment]
`@IfBuildProperty` removes the bean at build time If the property ist not set. I think you are looking for `@LookupIfProperty` (https://quarkus.io/guides/cdi-reference#declaratively-choose-beans-that-can-be-obtained-by-programmatic-lookup).

You can iterate over all instances with `Instance<Service>#stream()`.

**Comment by mschorsch** [comment, marked as answer]
You can check if an extension ist present with [Extension Capabilities](https://quarkus.io/guides/capabilities).

https://quarkus.io/guides/capabilities#querying-capabilities

```java
    @BuildStep
    HealthBuildItem addHealthCheck(Capabilities capabilities, DataSourcesBuildTimeConfig dataSourcesBuildTimeConfig) {
        if (capabilities.isPresent(Capability.SMALLRYE_HEALTH)) {
            return new HealthBuildItem("io.quarkus.agroal.runtime.health.DataSourceHealthCheck",
                    dataSourcesBuildTimeConfig.healthEnabled);
        } else {
            return null;
        }
    }
```
